specification/communication/data-plane/role/preview/2022-01-01-preview/CommunicationRole.json

```diff
@@ -0,0 +1,439 @@
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get a Communication Role": {
+            "$ref": "./examples/GetRole.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CommunicationRoleCreatePermissions": {
+      "description": "The object containing the various types of permissions.",
+      "required": [
+        "dataActions"
+      ],
+      "type": "object",
+      "properties": {
+        "dataActions": {
+          "description": "The list of data oriented action permissions.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CommunicationRolePermissionActions"
+          }
+        }
+      }
+    },
+    "CommunicationRoleCreateRequest": {
+      "required": [
+        "displayName",
+        "isEnabled",
+        "permissions"
+      ],
+      "type": "object",
+      "properties": {
+        "displayName": {
+          "description": "The human readable name of the Role.",
+          "type": "string"
+        },
+        "permissions": {
+          "$ref": "#/definitions/CommunicationRoleCreatePermissions"
+        },
+        "isEnabled": {
+          "description": "Is Role enabled for assignments.",
+          "type": "boolean"
+        }
+      }
+    },
+    "CommunicationRoleGetAllResponse": {
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The query response content",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CommunicationRoleResponse"
+          }
+        },
+        "maxItems": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "@nextLink": {
+          "type": "string"
+        }
+      }
+    },
+    "CommunicationRolePermissionActions": {
+      "description": "The available action permissions list.",
+      "enum": [
+        "communication.services/calling/audio/receive/action",
+        "communication.services/calling/video/receive/action",
+        "muteThemselves",
+        "communication.services/calling/mute/others/audio/action",
+        "communication.services/calling/mute/others/video/action",
+        "communication.services/calling/audio/send/action",
+        "communication.services/calling/video/send/action",
+        "communication.services/calling/screenshare/send/action",
+        "communication.services/calling/mute/others/screenshare/action",
+        "communication.services/calling/recording/action",
+        "communication.services/calling/transcript/action",
+        "communication.services/calling/livecaptions/action",
+        "communication.services/calling/admitLobby/action",
+        "communication.services/calling/participants/add/action",
+        "communication.services/calling/participants/remove/action",
+        "communication.services/calling/managePSTNParticipants/action",
```

Comment:
Have we considered to split this into add and remove to be more consistent with /participants?

Maybe
```
"communication.services/calling/pstnParticipants/add/action",
"communication.services/calling/pstnParticipants/remove/action",
```

Reply:
Please forward those comments in the [spec document](https://microsoft.sharepoint-df.com/teams/IC3SDK/Shared%20Documents/Roles/ACS%20-%20Roles%20and%20Permissions%20-%20Architecture%20Specification.docx?web=1).